pkg/trace/agent/agent.go

```diff
@@ -269,6 +272,25 @@ func (a *Agent) Process(p *api.Payload) {
 				sampler.SetPreSampleRate(root, rate)
 			}
 		}
+
+		if a.conf.ErrorTracking.Only {
+			var keep []*pb.Span
+			for _, span := range chunk.Spans {
+				if eligibleForErrorTracking(span) {
+					span.Meta[tagErrorTrackingMode] = "1"
+					keep = append(keep, span)
+					break
+				}
+			}
+			if len(keep) == 0 {
+				p.RemoveChunk(i)
+				continue
+			}
+			// only keep eligible spans
+			chunk.Spans = keep
```

Comment:
doing this there are good chances we'll remove the root span. Doesn't seem too problematic from what I saw in APM when testing, but maybe there are deeper implications, on sampling for instance.
We could decide to always keep the root span when there are errors.